tech-jobs/summary_stats/scripts/entryProcessor.py

```diff
@@ -0,0 +1,55 @@
+import re
+import itertools
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import unicodecsv as csv
+import seaborn as sns
+from collections import Counter
+from cycler import cycler
+from textwrap import wrap
+
+class EntryProcessor:    
+    
+    def __init__(self, employee, records):
+        self.current_id = None
+        self.employee = employee
+        self.records = records
+    
+    # When id changes, record a possible leave and then clear contents.
+    def reinitialize(self,entry):
+        if self.employee.f_current == 'False':
+            self.records.leave_record(self.employee, 'unknown') #data type of industry? 
+        self.employee.reset([entry[x] for x in [1,2,3,5,8,9,10,29]], entry[6])
+        self.current_id = entry[0]
```

Comment:
It is probably faster, and definitely cleaner from a coding-style perspective, to simply instantiate a new struct for each employee, rather than reusing and clearing one object.

Reply:
Fixed.